connectivity/drivers/nfc/PN512/CMakeLists.txt

```diff
@@ -26,3 +28,11 @@ target_sources(mbed-nfc
         source/transceiver/pn512_timer.c
         source/transceiver/pn512_transceive.c
 )
+
+target_link_libraries(mbed-nfc PRIVATE mbed-nfc-pn512)
+
+# Expose include directories in case applications need it
```

Comment:
It isn't immediately obvious why we're doing this. The `INTERFACE_` properties should be appended to non-`INTERFACE_` properties and forwarded if we use `target_link_libraries(PUBLIC)`. Here the consuming target links to the `mbed-nfc-pn512` target privately and then only exposes the include directories as public usage requirements of the consumer. Can we add an explanation in the commit message for why we need to do this?

Reply:
I wonder if commit message is sufficient. This will be in multiple places where we have drivers (current architecture how the libraries include things). We might remove this once we get add/remove library and refactor the include paths (how configuration from a target is exposed to a Mbed OS library or application - mbed.h and similar - exposing too much sometimes).

drivers are interfaces, extending the main object library like mbedtls/cellular or ble. As they however provide headers that configure the main library and also library consumers (mbedtls driver provides configuration that needs to be exposed to consumers of mbedtls like cellular, application etc).  Only this configuration (headers) shall be exposed, not object files as they are part of the main library (duplications errors otherwise).

Reply:
Do we have the rationale for this workaround documented anywhere? If we're repeating this pattern throughout mbed-os we should add something to our CMake design document, if we haven't already (I couldn't find anything when I checked it).

Reply:
That is my point, we should add it.

@hugueskamba Can you add it to the  feature branch?